lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -80,6 +89,31 @@ static void addCanonicalizerMethod(OwningRewritePatternList &results,
   results.insert<Folder>(context);
 }
 
+/// Performs constant folding `calculate` with element-wise behavior on the two
+/// attributes in `operands` and returns the result if possible, with the
+/// specified result type.
+///
+/// TODO: Maybe the upstream constFoldBinaryOp can be made to have a
+/// configurable result type as well.
+template <class AttrElementT,
+          class ElementValueT = typename AttrElementT::ValueType,
+          class CalculationT = function_ref<bool(ElementValueT, ElementValueT)>>
+Attribute constFoldBinaryRelationalOp(ArrayRef<Attribute> operands,
```

Comment:
Plz mark `static`

Reply:
I think that `ElementValueT` will always be `IntegerAttr` since that is what all primitives work on.  If so, I'd de-generisize that

Reply:
Good points, will do.

Reply:
Done